Cesium3DTilesSelection/src/TilesetLoadIonAssetEndpoint.cpp

```diff
@@ -60,6 +60,9 @@ struct Tileset::LoadIonAssetEndpoint::Private {
           assert(false);
         });
   }
+
+  static std::unordered_map<std::string, std::shared_ptr<IAssetRequest>>
+      endpointCache;
```

Comment:
We shouldn't hold onto the IAssetRequest instance like this. The problem is that IAssetRequest is implemented by cesium-native's client (e.g. Cesium for Unreal or O3DE or whatever) and may (probably does) hold some of the client's resources hostage. For example, maybe there's a limit on the number of requests that can exist at once, and these requests will count against that limit. Or maybe it's really important that request resources be released before, say, the HTTP module shuts down, but we can't do that if a static field is keeping them alive.

So it may be possible to just cache the parts of the request/response that are needed. Or, if necessary, you can even create another class that implements IAssetRequest and store instances of that class in the cache instead. We do something similar in CachingAssetAccessor.cpp.